vault/auth.go

```diff
@@ -182,7 +183,16 @@ func (c *Core) enableCredentialInternal(ctx context.Context, entry *MountEntry,
 	if backendType != logical.TypeCredential {
 		return fmt.Errorf("cannot mount %q of type %q as an auth backend", entry.Type, backendType)
 	}
-
+	// update the entry running version with the backend's reported version
+	if versioner, ok := backend.(logical.Versioner); ok {
+		entry.RunningVersion = versioner.Version().Version
+	}
+	if entry.RunningVersion == "" {
+		// don't set the running version to a builtin if it is running as an external plugin
+		if externaler, ok := backend.(logical.Externaler); !ok || !externaler.IsExternal() {
```

Comment:
When would `ok` be false, is that more a defensive check to avoid unexpected panics?

Reply:
It's mostly defensive. There are so many plugin middlewares we use, and if we end up adding a new one in the future, I wouldn't want this to panic.

Reply:
Gotcha, seems prudent 👍